### PR TITLE
DI API review changes

### PR DESCRIPTION
**Pull request by BrennanConroy**
#185
#189
#190
#191
#192

### CODE REVIEW

File: src/Microsoft.Framework.DependencyInjection.Interfaces/IServiceCollection.cs [hunk 1]
```diff
@@ -8,13 +8,13 @@ namespace Microsoft.Framework.DependencyInjection
...
-    public interface IServiceCollection : ICollection<ServiceDescriptor>
+    public interface IServiceCollection : IList<ServiceDescriptor>
     {
         /// <summary>
         /// Adds the <paramref name="descriptor"/> to this instance.
         /// </summary>
         /// <param name="descriptor">The <see cref="ServiceDescriptor"/> to add.</param>
         /// <returns>A reference to the current instance of <see cref="IServiceCollection"/>.</returns>
-        new IServiceCollection Add(ServiceDescriptor descriptor);
+        //new IServiceCollection Add(ServiceDescriptor descriptor);
```
Comment (davidfowl) [inline src/Microsoft.Framework.DependencyInjection.Interfaces/IServiceCollection.cs]: Delete everything?

File: src/Microsoft.Framework.DependencyInjection/ServiceCollection.cs [hunk 2]
```diff
@@ -20,14 +21,20 @@ public class ServiceCollection : IServiceCollection
...
+        public ServiceDescriptor this[int index]
         {
-            _descriptors.Add(descriptor);
-            return this;
+            get { return _descriptors[index]; }
+            set { _descriptors.Add(value); }
         }
 
         /// <inheritdoc />
+        //public IServiceCollection Add([NotNull] ServiceDescriptor descriptor)
```
Comment (davidfowl) [inline src/Microsoft.Framework.DependencyInjection/ServiceCollection.cs]: Remove teh dead code.

File: src/Microsoft.Framework.DependencyInjection/ServiceCollection.cs [hunk 3]
```diff
@@ -20,14 +21,20 @@ public class ServiceCollection : IServiceCollection
...
         public bool IsReadOnly => false;
 
-        /// <inheritdoc />
-        public IServiceCollection Add([NotNull] ServiceDescriptor descriptor)
+        public ServiceDescriptor this[int index]
         {
-            _descriptors.Add(descriptor);
-            return this;
+            get { return _descriptors[index]; }
+            set { _descriptors.Add(value); }
```
Comment (davidfowl) [inline src/Microsoft.Framework.DependencyInjection/ServiceCollection.cs]: Whatttt?

File: src/Microsoft.Framework.DependencyInjection/ServiceCollection.cs (hunk 2 above)
Comment (BrennanConroy) [inline src/Microsoft.Framework.DependencyInjection/ServiceCollection.cs]: Oh thanks, forgot about that

File: test/Microsoft.Framework.DependencyInjection.Tests/ServiceCollectionExtensionTest.cs [hunk 4]
```diff
@@ -202,21 +202,21 @@ public void TryAddWithEnumerableReturnsFalseIfNoneAdded()
...
-            // Act
-            collection.AddTypeActivator();
-
-            // Assert
-            var descriptor = Assert.Single(collection);
-            Assert.Equal(typeof(ITypeActivator), descriptor.ServiceType);
-            Assert.Null(descriptor.ImplementationInstance);
-            Assert.Equal(ServiceLifetime.Singleton, descriptor.Lifetime);
-        }
+        //[Fact]
```
Comment (davidfowl) [inline test/Microsoft.Framework.DependencyInjection.Tests/ServiceCollectionExtensionTest.cs]: Remove dead code.

**Comment by davidfowl** [comment]
Remove dead code

**Comment by davidfowl** [comment]
:shipit:

**Pull request merged**